helper/keysutil/lock_manager.go

```diff
@@ -54,27 +54,85 @@ type PolicyRequest struct {
 }
 
 type LockManager struct {
-	useCache bool
-	// If caching is enabled, the map of name to in-memory policy cache
-	cache sync.Map
-
-	keyLocks []*locksutil.LockEntry
+	useCache  bool
+	cacheType CacheType
+	cache     Cache
+	keyLocks  []*locksutil.LockEntry
+	lock      sync.RWMutex
 }
 
 func NewLockManager(cacheDisabled bool) *LockManager {
 	lm := &LockManager{
-		useCache: !cacheDisabled,
-		keyLocks: locksutil.CreateLocks(),
+		useCache:  !cacheDisabled,
+		cacheType: NotInitialized,
+		keyLocks:  locksutil.CreateLocks(),
+		lock:      sync.RWMutex{},
 	}
+	lm.ConvertCacheToSyncmap()
 	return lm
 }
 
+func (lm *LockManager) ConvertCacheToLRU(size int) error {
+	lm.lock.Lock()
+	defer lm.lock.Unlock()
+	// update LockManager
+	newCache, err := NewTransitLRU(size)
+	if err != nil {
+		return err
+	}
+	lm.cacheType = LRU
+	lm.cache = newCache
+	return nil
+}
+
+func (lm *LockManager) ConvertCacheToSyncmap() {
+	lm.lock.Lock()
+	defer lm.lock.Unlock()
+	// update LockManager
+	lm.cacheType = SyncMap
+	lm.cache = NewTransitSyncMap()
+}
+
+func (lm *LockManager) GetCacheType() CacheType {
+	// Locked because lm.cache could change
+	lm.lock.RLock()
+	defer lm.lock.RUnlock()
+	return lm.cacheType
+}
+
+func (lm *LockManager) GetCacheSize() int {
+	// Locked because lm.cache could change
+	lm.lock.RLock()
+	defer lm.lock.RUnlock()
+	return lm.cache.Size()
+}
```

Comment:
I think we can remove most (or all?) of these functions now that `SetCacheSize` exists, right?

Reply:
Or at least make them non-exported

Reply:
Still making use of them but they certainly don't need to be exported